### PR TITLE
[change] Update react-native-web example

### PR DESCRIPTION
**Pull request by dcalhoun**
Utilize ES modules, which are now the default export for
`react-native-web`.

_NOTE:_ [This example requires `next@^7.0.0`](https://bit.ly/2PaEhao).

### CODE REVIEW

File: examples/with-react-native-web/next.config.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const esNodeModules = /react-native-web(?!.*node_modules)/
+const cjsNodeModules = /node_modules\/(?!react-native-web(?!.*node_modules))/
+
+module.exports = {
+  webpack: (config, { buildId, dev, isServer, defaultLoaders }) => {
+    // Alias all `react-native` imports to `react-native-web`
+    config.resolve.alias = {
+      'react-native$': 'react-native-web'
```
Comment (HaNdTriX) [inline examples/with-react-native-web/next.config.js]: Lets drop this and fix this in the code instead.
https://github.com/zeit/next.js/blob/canary/examples/with-react-native-web/pages/index.js#L2

**Review by HaNdTriX — CHANGES_REQUESTED** [review]
We might want to use the following plugin instead:

**Example:** `next.config.js`

```js
const withTM = require('next-plugin-transpile-modules')

module.exports = withTM({
  transpileModules: ['react-native-web']
})
```

[next-plugin-transpile-modules](https://github.com/martpie/next-plugin-transpile-modules)

File: examples/with-react-native-web/next.config.js (hunk 1 above)
Comment (dcalhoun) [reply examples/with-react-native-web/next.config.js]: Making that change goes against the [primary goal](https://github.com/necolas/react-native-web#contributing) of `react-native-web`.

>The main purpose of this repository is to help evolve React web and native development towards the platform-agnostic design of React Native

The primary goal of `react-native-web` is to leverage the _platform-agnostic_ design of React Native. Having a file specifically import `react-native-web` would prevent using that file on native platforms.

I strongly recommend not removing this webpack alias. It is a part of the [standard "Getting started" guide](https://github.com/necolas/react-native-web/blob/master/docs/guides/getting-started.md#configuring-a-module-bundler) provided by `react-native-web`.

**Comment by dcalhoun** [comment]
@HaNdTriX if the Next.js team would prefer to use `next-plugin-transpile-modules` in place of the custom webpack config I've included, that is fine. Just let me know.

**Comment by dcalhoun** [comment]
@HaNdTriX this should be ready for merge now.

**Review by HaNdTriX — APPROVED** [review]

**Pull request merged**